src/box/txn.h

```diff
@@ -893,6 +893,38 @@ txn_is_nop(const struct txn *txn)
 	return txn->n_new_rows + txn->n_applier_rows == 0;
 }
 
+static inline bool
+txn_is_fully_temporary(struct txn *txn)
```

Comment:
Nit: the txn can be `const` here, no? Same for `txn_is_fully_remote()`.

Reply:
No, unfortunately, using the `stailq` API requires the original struct to be non-const:
```console
/Users/glebedev/Work/tarantool/src/box/txn.c:671:2: error: passing 'const struct stailq *' to parameter of type 'struct stailq *' discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
        stailq_foreach_entry(stmt, &txn->stmts, next) {

```